### PR TITLE
Add fetching from remote

### PR DESCRIPTION
**Pull request by bigdavedev**
References issue #137

Signed-off-by: Dave Brown <d.brown@bigdavedev.com>

### CODE REVIEW

**Comment by bigdavedev** [comment]
I am trying to implement a dialog that will allow you to choose from multiple remotes, should you have that configured. I believe that would suffice for the majority of SGit's users.  Fetching from specific refspecs would be best left as a feature request later on.  

To summarize, I intend to deliver the following:
- [x] Fetch from origin (default)
- [ ] Fetch from selected remote
- [ ] Fetch all remotes

**Comment by bigdavedev** [comment]
@maks I consider this now ready for review (tearing apart). I welcome any suggestions for improvements that you may have.

**Comment by maks** [comment]
sorry for the delay @bigdavedev , I've started looking through it, looks good so far and not too big a piece of new code, but I'll need to get back to finish going through it and trying out running with the change in the next day or 2.

File: src/me/sheimi/sgit/repo/tasks/repo/FetchTask.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        }
+
+        final FetchCommand fetchCommand = git.fetch()
+                .setProgressMonitor(new BasicProgressMonitor())
+                .setTransportConfigCallback(new SgitTransportCallback())
+                .setRemote(remote);
+
+        final String username = mRepo.getUsername();
+        final String password = mRepo.getPassword();
+        if (username != null && password != null && !username.equals("")
```
Comment (maks) [inline src/me/sheimi/sgit/repo/tasks/repo/FetchTask.java]: would isEmpty() be a bit to use?

**Comment by maks** [comment]
@bigdavedev the code all looks good to me (minus 1 tiny quibble) but when I tested this by using this repo and then adding your fork and choosing to fetch from just your fork, it doesn't seem to work correctly as I don't see your branches in the list of branches in the apps UI, even though the fetch operation seems to complete fine without any errors. 
Would you be able to see if you can reproduce this?  (I tested with the SDK emulator, API 23)

**Comment by maks** [comment]
@bigdavedev replying to your comment https://github.com/sheimi/SGit/issues/189#issuecomment-225338916

Yes I am literally looking into it more myself now. You're right all the refs appear to be there  and have the correct SHAs when I look in .git/refs but they don't even show up in the branches list UI, so I wonder if its also a problem with dealing with refs in subdirs. I'll dig into that UI code later on this weekend and see if I can figure out whats wrong, but this is definitely a separate issue to your PR (I'll raise a sep issue for it) and I'm fine to merge the PR.

**Pull request merged**